test/suites/greenboot.robot

```diff
@@ -0,0 +1,123 @@
+Resource            ../resources/ostree-health.resource
+Library    SSHLibrary
+Suite Setup         Setup Suite 
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${LOCAL_CONFIG_FILE}                  ../docs/config/busybox_running_check.sh
+${REMOTE_CONFIG_FILE}                 /etc/greenboot/check/required.d/50_busybox_running_check.sh
+${GREENBOOT_CONFIG_FILE}              /etc/greenboot/greenboot.conf
+${GREENBOOT_CONFIG_CONTENT}           MICROSHIFT_WAIT_TIMEOUT_SEC=180
+${MANIFEST_SOURCE_DIR}                assets/kustomizations/greenboot/
+${MANIFEST_DIR}                       /etc/microshift/manifests
+${MICROSHIFT_CONFIG}                  /etc/microshift/config.yaml
+*** Test Cases ***
+
+# user-workload-health
+
+GreenbootHealth Run with User Workload
+    [Documentation]    check
+    Upload User Config
+    Microshift Cleanup And Start
+    Cleanup User Config
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Microshift Disrupt Service
+    Run Keyword And Expect Error    	0 != 1
+    ...    Microshift Cleanup And Start
+    Microshift Restore Service
+    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift Pod Network failure
+    Microshift Disrupt Pod Network
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+    Microshift Cleanup And Start
+
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Update Greenboot Config
+
+Update Greenboot Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Upload String To File   ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE}
+
+Upload User Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Put File    ${LOCAL_CONFIG_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${REMOTE_CONFIG_FILE} ] || sudo ln -s /tmp/busybox_running_check.sh ${REMOTE_CONFIG_FILE} 
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Put Directory    ${MANIFEST_SOURCE_DIR}     /tmp/manifests
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp -f /tmp/manifests/* /etc/microshift/manifests/ && sudo rm -rf /tmp/manifests
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Cleanup User Config
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${REMOTE_CONFIG_FILE} ; sudo rm -rf /etc/microshift/manifests/*
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Greenboot Healthy
+    [Documentation]    check
+    Greenboot Restart
+    Wait Until Greenboot Health Check Exited
+
+Microshift Disrupt Service
+    ${stdout}    ${rc}=    Execute Command
+    ...    dnf remove -y hostname
```

Comment:
This assumes non-ostree OS. We need to have a condition here for the code to run `rpm-ostree` command instead of `dnf`. 
Alternatively, we can choose to `sudo chmod 000 $(which hostname)`. It can also work on ostree systems after `sudo rpm-ostree usroverlay`

Reply:
This is [how you can detect](https://github.com/openshift/microshift/blob/main/packaging/greenboot/microshift_set_healthy.sh#L13) if it's an ostree system.

Let's add a function to resources/ostree.resource?

Reply:
i like the `chmod 000`  idea better and it doesnt require fiddling with RPMs

Reply:
handled in https://github.com/openshift/microshift/pull/2213/commits/1abb4ea635acac6b3073c67e0047e8fcfa7e2881